homeassistant/components/intellifire/config_flow.py

```diff
@@ -4,60 +4,132 @@
+                    errors["base"] = "already_discovered"
+                    LOGGER.debug("All discovered fireplaces have been configured")
+
+        else:  # There was User Input
+
+            placeholder = {CONF_HOST: user_input[CONF_HOST]}
+            try:
+                serial = await validate_host_input(user_input[CONF_HOST])
+            except (ConnectionError, ClientConnectionError):
+                errors["base"] = "cannot_connect"
+                local_schema = vol.Schema(
+                    {vol.Required(CONF_HOST, default=user_input[CONF_HOST]): str}
+                )
+            else:
+                await self.async_set_unique_id(serial)
+
+                # check if found before
+                self._abort_if_unique_id_configured(
+                    updates={
+                        CONF_HOST: user_input[CONF_HOST],
+                    }
+                )
+
+                return self.async_create_entry(
+                    title="Fireplace",
+                    data={
+                        CONF_HOST: user_input[CONF_HOST],
+                    },
+                )
+
+        return self.async_show_form(
+            step_id="local_config",
+            errors=errors,
+            description_placeholders=placeholder,
+            data_schema=local_schema,
+        )
+
+    async def async_step_pick_device(
         self, user_input: dict[str, Any] | None = None
     ) -> FlowResult:
-        """Handle the initial step."""
-        if user_input is None:
-            return self.async_show_form(
-                step_id="user", data_schema=STEP_USER_DATA_SCHEMA
-            )
-        errors = {}
+        """Pick which device to configure."""
+
+        if user_input is not None:
+            return await self.async_step_local_config(user_input=user_input)
 
-        try:
-            serial = await validate_input(self.hass, user_input[CONF_HOST])
-        except (ConnectionError, ClientConnectionError):
-            errors["base"] = "cannot_connect"
-        else:
-            await self.async_set_unique_id(serial)
-            self._abort_if_unique_id_configured(
-                updates={CONF_HOST: user_input[CONF_HOST]}
-            )
-
-            return self.async_create_entry(
-                title="Fireplace",
-                data={CONF_HOST: user_input[CONF_HOST]},
-            )
         return self.async_show_form(
-            step_id="user", data_schema=STEP_USER_DATA_SCHEMA, errors=errors
+            step_id="pick_device",
+            data_schema=vol.Schema(
+                {vol.Required(CONF_HOST): vol.In(self._discovered_hosts)}
+            ),
         )
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Start the user flow."""
+
+        if self._discovered_hosts == []:
+            await self._find_fireplaces()
+        return await self.async_step_local_config()
```

Comment:
If you discover fireplaces, move to the `async_step_pick_device` step. 

If you don't discovery any fireplaces, move to a manual entry step.